src/test/java/teammates/test/cases/datatransfer/AdminEmailAttributesTest.java

```diff
@@ -0,0 +1,187 @@
+        date = new Date();
+        fieldValidator = new FieldValidator();
+
+        adminEmailAttributes = new AdminEmailAttributes(
+                subject, addressReceiverListString, groupReceiverListFileKey, content, date);
+
+        adminEmailAttributes.createDate = new Date();
+    }
+
+    @Test
+    public void testValidate() throws Exception {
+
+        // Valid input
+        assertTrue("Valid input", adminEmailAttributes.isValid());
+        List<String> errorList = adminEmailAttributes.getInvalidityInfo();
+        assertTrue("Valid input should return an empty list of errors", errorList.isEmpty());
+
+        // Content empty
+        Text emptyContent = new Text("");
+        AdminEmailAttributes invalidAttributesContentEmpty = new AdminEmailAttributes(
+                subject, addressReceiverListString, groupReceiverListFileKey, emptyContent, date);
+
+        String expectedContentEmptyError = getPopulatedErrorMessage(
+                FieldValidator.EMAIL_CONTENT_ERROR_MESSAGE, invalidAttributesContentEmpty.getContentValue(),
+                FieldValidator.EMAIL_CONTENT_FIELD_NAME, FieldValidator.REASON_EMPTY,
+                0);
+
+        assertEquals("Invalid content input should return appropriate error string", expectedContentEmptyError,
+                StringHelper.toString(invalidAttributesContentEmpty.getInvalidityInfo()));
+
+        String expectedStringContentError = fieldValidator.getInvalidityInfoForEmailContent(emptyContent);
+        assertEquals(FieldValidator.EMAIL_CONTENT_ERROR_MESSAGE, expectedStringContentError);
+
+        assertFalse("Invalid input", invalidAttributesContentEmpty.isValid());
+
+        // Subject exceeds max length
+        String veryLongSubj = StringHelperExtension.generateStringOfLength(FieldValidator.EMAIL_SUBJECT_MAX_LENGTH + 1);
+        AdminEmailAttributes invalidAttributesSubjectLength = new AdminEmailAttributes(
+                veryLongSubj, addressReceiverListString, groupReceiverListFileKey, content, date);
+
+        String expectedEmptySubjectLengthError = getPopulatedErrorMessage(
+                FieldValidator.SIZE_CAPPED_NON_EMPTY_STRING_ERROR_MESSAGE, invalidAttributesSubjectLength.getSubject(),
+                FieldValidator.EMAIL_SUBJECT_FIELD_NAME, FieldValidator.REASON_TOO_LONG,
+                FieldValidator.EMAIL_SUBJECT_MAX_LENGTH);
+
+        assertEquals("Invalid subject input should return appropriate error string",
+                expectedEmptySubjectLengthError, StringHelper.toString(invalidAttributesSubjectLength.getInvalidityInfo()));
+
+        // Subject must start with alphanumeric character, cannot contain vertical bar(|) or percent sign(%).
+        String invalidSubjectChars = "%Invalid%Subject|";
+        AdminEmailAttributes invalidAttributesSubjectChars = new AdminEmailAttributes(
+                invalidSubjectChars, addressReceiverListString, groupReceiverListFileKey, content, date);
+
+        String expectedError =
```

Comment:
- This error message should be able to be generated with `getPopulatedErrorMessage`, as what is done for others.
- `expectedError` is vague; expected what kind of error?